DSD-524: Chakra Notification

EdwinGuzman opened the pull request:
Fixes JIRA ticket [DSD-524](https://jira.nypl.org/browse/DSD-524)

## This PR does the following:
- Updates the `Notification` component to use Chakra.
- Note: I removed the export for the `NotificationContent` and `NotificationHeading` components and they are now props instead.

## How has this been tested?

Storybook.

### Checklist:

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

<!--- This step is done AFTER creating a PR. -->
<!--- Tugboat creates a static Storybook preview URL once the PR is created. -->
<!--- Copy the URL to the relevant Storybook page here. -->

- [ ] View [the example in Storybook]()

EdwinGuzman committed:
Updating the Notification component to use Chakra

EdwinGuzman committed:
Merge branch 'development' into DSD-524/chakra-notification

* development:
  added entry for Breadcrumbs to CHANGELOG
  prep for release v0.25.1
  0.25.1

EdwinGuzman committed:
Updating the changelog

bigfishdesign13 commented on src/theme/components/notification.ts:
```diff
@@ -0,0 +1,95 @@
+    }
+    return {
+      display: "flex",
+      marginBottom: "s",
```
Change this to `xxs`.  As I look at this in the browser, I am not sure why this margin was originally set as `s`.  That's too large a space.

bigfishdesign13 commented on src/components/Button/Button.tsx:
```diff
@@ -26,6 +26,8 @@ interface ButtonProps {
   /** The action to perform on the `<button>`'s onClick function */
   onClick?: (event: React.MouseEvent | React.KeyboardEvent) => void;
+  /** Optionally pass in additional Chakra-based styles. */
+  sx?: { [key: string]: any };
```
Why were these changes included in this PR?

EdwinGuzman replied:
Because the `Button` component inside the `Notification` component has updated styles and I'm passing those through this prop - https://github.com/NYPL/nypl-design-system/pull/724/files#diff-1019e818df12ccb2db0eafad4b181840c4eeecd95b057fe177466b21a68d7b54R129

But, as mentioned in an earlier PR, the "sx" name is not the best so I'll update this.

bigfishdesign13 replied:
OK, got it.

bigfishdesign13 commented on src/components/Notification/Notification.tsx:
```diff
@@ -1,201 +1,183 @@
-  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
-  modifiers?: string[];
+  /** Content to be rendered in a `NotificationContent` component. */
+  notificationContent: JSX.Element;
```
Why `JSX.Element` for this and `notificationHeading`?  Why not define them as `string`?

EdwinGuzman replied:
Right, I wasn't sure what the right type would be and maybe it can be `string | JSX.Element`.  All the previous examples were plain strings but what if more complicated (such as <b>, <i>, or links) elements should be rendered?

bigfishdesign13 replied:
Right, `notificationContent` might have more sophisiticated content, so allowing for `JSX.Element` does make sense.  But I think `notificationHeading` can just be `string`.

bigfishdesign13 replied:
I see that `Hero` uses the `string | JSX.Element` pattern, so let's do the same here.  More often than not, the actual content for this section is going to be a plain string.

bigfishdesign13 reviewed (changes requested):
Another complicated conversion.  Overall, it works well.  I have two questions and one requested change.

EdwinGuzman committed:
Merging from development and updating based on feedback

EdwinGuzman commented:
Updated.

EdwinGuzman commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr724-u1ernhaulbukwkf8kdly8hhkuxytjbpd.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/616891f11ff56a6b8be90167

bigfishdesign13 reviewed (changes requested):
Let's go ahead and change the assignment of `notificationContent` to `string | JSX.Element` and bring it into alignment with what had been done in `Hero`.

EdwinGuzman committed:
Updating the sx prop to additionalStyles

EdwinGuzman commented:
Updated and also updated the `sx` prop name to `additionalStyles`. I think that makes it clearer that it's a prop for the DS components whereas the Chakra components use `__css` or `sx`.

EdwinGuzman committed:
Making notificationContent type string or JSX element

EdwinGuzman commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr724-u1ernhaulbukwkf8kdly8hhkuxytjbpd.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/616891f11ff56a6b8be90167

bigfishdesign13 reviewed (approved):
Very nice.

EdwinGuzman committed:
Merge branch 'development' into DSD-524/chakra-notification

* development:
  fixed spacing issue for Hero Tertiary

EdwinGuzman committed:
Merging from development and fixing conflicts
